responder/msplanner/msplanner.go

```diff
@@ -0,0 +1,196 @@
+	"github.com/microsoftgraph/msgraph-sdk-go/models"
+	"github.com/microsoftgraph/msgraph-sdk-go/models/odataerrors"
+	"github.com/microsoftgraph/msgraph-sdk-go/planner/tasks/item"
+)
+
+type MSPlannerTask struct {
+	Title       string
+	PlanID      string
+	Priority    int32
+	Description string
+	BucketID    string
+}
+
+type PlanConfig struct {
+	ID         string         `json:"id"`
+	Name       string         `json:"name"`
+	Buckets    []PlanBucket   `json:"buckets"`
+	Priorities map[string]int `json:"priorities"`
+}
+
+type PlanBucket struct {
+	ID   string `json:"id"`
+	Name string `json:"name"`
+}
+
+type MSPlannerClient struct {
+	client  *msgraphsdk.GraphServiceClient
+	groupID string
+}
+
+func NewClient() (MSPlannerClient, error) {
+	cred, err := azidentity.NewDeviceCodeCredential(&azidentity.DeviceCodeCredentialOptions{
+		ClientID: "CLIENT_ID",
+		UserPrompt: func(ctx context.Context, message azidentity.DeviceCodeMessage) error {
+			fmt.Println(message.Message)
+			return nil
+		},
+	})
+	if err != nil {
+		return MSPlannerClient{}, fmt.Errorf("Error creating credentials: %v\n", err)
+	}
+
+	auth, err := kiotaAuth.NewAzureIdentityAuthenticationProviderWithScopes(cred, []string{"User.Read"})
+	if err != nil {
+		return MSPlannerClient{}, fmt.Errorf("Error authentication provider: %v\n", err)
+	}
+
+	adapter, err := msgraphsdk.NewGraphRequestAdapter(auth)
+	if err != nil {
+		return MSPlannerClient{}, fmt.Errorf("Error creating adapter: %v\n", err)
+	}
+	client := msgraphsdk.NewGraphServiceClient(adapter)
+	return MSPlannerClient{client: client}, nil
+}
+
+func (c MSPlannerClient) CreateTask(opts MSPlannerTask) (models.PlannerTaskable, error) {
+	body := models.NewPlannerTask()
+
+	// body.SetPriority()
+	// TODO: Description is a hell-hole
+	// Use models.PLannerTaskDetails and then set
+	body.SetPlanId(&opts.PlanID)
+	body.SetBucketId(&opts.BucketID)
+	body.SetTitle(&opts.Title)
+	body.SetPriority(&opts.Priority)
+
+	result, err := c.client.Planner().Tasks().Post(body)
+	return result, openDataError(err)
+}
+
+func (c MSPlannerClient) AddComment(taskID, comment string) error {
+
+	// First check if conversation thread already exists for not, if not create new
+	// Get task
+	// if task.GetConvoId == nil
+
+	task, err := c.client.Planner().TasksById(taskID).Get()
+	if err != nil {
+		return openDataError(err)
+	}
+
+	post := getCommentPost(comment)
+
+	// Create a new conversation thread if task does not has one
+	if task.GetConversationThreadId() == nil {
+		convBody := models.NewConversationThread()
+		topic := fmt.Sprintf("Conversation thread topic for taskID: %s", taskID)
+		convBody.SetTopic(&topic)
+
+		convBody.SetPosts([]models.Postable{post})
+
+		result, err := c.client.GroupsById(c.groupID).Threads().Post(convBody)
+		if err != nil {
+			fmt.Println("error creating Conversation thread")
+			return openDataError(err)
+		}
+
+		// TODO: For debugging
+		fmt.Println("Conversation thread created")
+		fmt.Println(result)
```

Comment:
use commons/logger trace

Reply:
Yeah, this is just for debugging on local, will remove all print statements